src/software/jetson_nano/rotary_encoder/RotaryEncoder.py

```diff
@@ -0,0 +1,141 @@
+import Jetson.GPIO as GPIO
+
+"""
+The state of a rotary encoder is dependent on the values read on pin A and pin B.
+The input read on pin A and pin B can either be 0 or 1 and therefore we have four
+possible states. The encoder will transition through these four states during a 
+single rotation and we can use these state transitions to determine whether the encoder 
+has been rotated clockwise or counterclockwise.
+
+Channel B and A go through 4 states for a single rotation.
+The following are values for clockwise rotation:
+  Channel B Channel A
+     0          0
+     0          1
+     1          1
+     1          0
+
+The defined states below are tuples where the first index is the value read from pin B and
+the second index the value read from pin A.
+"""
+STATE_0 = (0, 0)
+STATE_1 = (0, 1)
+STATE_2 = (1, 1)
+STATE_3 = (1, 0)
```

Comment:
Why not make this a list, e.g. `STATES = [(0,0), ...]`, then NUM_STATES can be replaced with len(STATES)? State transitions can also be (i +/- 1)%4 I think

Reply:
I have updated the file to use a list as you described and replaced the NUM_STATES variable with len(STATES). I decided to leave the state transitions as is because I am not sure if your method provides benefits as well as it may require some additional changes, e.g. direction variable, state machine to determine if clockwise or counterclockwise state change. However, I may be overthinking the implement ion of this. Let me know if you think there are benefits to using your method and I will try to make the changes.